lua/dap.lua

```diff
@@ -166,6 +166,7 @@ local DAP_QUICKFIX_CONTEXT = DAP_QUICKFIX_TITLE
 ---@field options nil|AdapterOptions
 ---@field enrich_config? fun(config: Configuration, on_config: fun(config: Configuration))
 ---@field reverse_request_handlers? table<string, fun(session: Session, request: dap.Request)>
+---@field is_multiline fun(text: string)
```

Comment:
Can you add a documentation about `is_multiline` into doc/dap.txt, and provide some examples of it? Say how should one write this funtion for python adapter? Doesn't this function require some "context" (or some syntactic knowledge) to determine whether the input requires further lines or not?

Reply:
I found this https://github.com/mfussenegger/nvim-dap/pull/773#issuecomment-1341492471, but this is not enough. There would be many counterexamples, and I don't think the sole argument `text` (the last line) is enough.  It would need to take all the lines entered so far as well.

Reply:
Also the type annotation is not correct:
```diff
-is_multiline fun(text: string)
+is_multiline fun(text: string):boolean
```

Reply:
I guess you ask for support for something like this:
```
> function a() {
...
... }
```
in my current implementation it is not supported, I'll look into it soon...

Reply:
@wookayin I have looked at it again, and I think the current implementation is better.
The current implementation works like this: Any user input can be considered either a single line input, or an input that starts a multiline block. The function is_multiline differentiate between the two. Once you enter a multiline mode, you will stay there until a **blank line** is sent by the user, that is the reason why the context is not sent, because it's always the first line.
The alternative comes into place once you are inside a multiline mode, in that case for each input you query is_multiline to see if the user should continue to input lines.
I think that the first approach is much simpler and probably covers most cases needed, as I mentioned in the last comment, it doesn't cover cases in which you have blank lines inside your block, but I think there needs to be a good example to justify adding the feature.
I am mostly concerned about an invalid configuration of the is_multiline function that will make the user stuck in multiline mode, not able to exit, this is only a problem in the second implementation.